colcon_bundle/verb/_archive_generators.py

```diff
@@ -73,3 +61,88 @@ def generate_archive(install_base,
     os.remove(bundle_tar_path)
 
     logger.info('Archiving complete')
+
+
+def generate_archive_v2(install_base,
+                        staging_path,
+                        bundle_base,
+                        metadata_paths,
+                        dependencies_changed):
+    """
+    Generate bundle archive v2.
+
+    This archive is a tarfile that contains multiple compressed archives:
+
+    output.tar
+    |- version
+    |- dependencies.tar.gz
+    |- workspace.tar.gz
+    |- metadata.tar.gz
+
+    :param install_base: Directory with built artifacts from the workspace
+    :param staging_path: Directory where all dependencies have been installed
+    :param bundle_base: Directory to place the output of this function
+    :param metadata_paths: [str] paths to files which should be included
+    in the metadata archive
+    :param dependencies_changed: Boolean representing whether the staging path
+    needs to be re-archvied
+    """
+    logger.info('Archiving the bundle output')
+    print('Creating bundle archive V2...')
+
+    archive_tar_path = os.path.join(bundle_base, 'output.tar')
+    workspace_tar_gz_path = os.path.join(bundle_base, 'workspace.tar.gz')
+
+    # Install directory
+    workspace_staging_path = os.path.join(bundle_base, 'workspace_staging')
+    workspace_path = os.path.join(workspace_staging_path, 'workspace')
```

Comment:
workspace_path -> workspace_install_path? Also could we follow the convention of having install path at ./opt/install from BUNDLE_CURRENT_PREFIX so that all the overlays are consistent.